// react-bulma-companion/lib/Navbar/NavbarItem/NavbarItem.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = NavbarItem;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _Element = _interopRequireDefault(require("../../Element"));

var _excluded = ["children", "className", "active", "expanded", "hasDropdown", "hasDropup", "hoverable", "tab"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function NavbarItem(_ref) {
  var children = _ref.children,
      className = _ref.className,
      active = _ref.active,
      expanded = _ref.expanded,
      hasDropdown = _ref.hasDropdown,
      hasDropup = _ref.hasDropup,
      hoverable = _ref.hoverable,
      tab = _ref.tab,
      props = _objectWithoutProperties(_ref, _excluded);

  var classes = (0, _classnames["default"])('navbar-item', className, {
    'is-active': active,
    'has-dropdown': hasDropdown || hasDropup,
    'has-dropdown-up': hasDropup,
    'is-hoverable': hoverable,
    'is-expanded': expanded,
    'is-tab': tab
  });
  return /*#__PURE__*/_react["default"].createElement(_Element["default"], _extends({
    className: classes
  }, props), children);
}

NavbarItem.propTypes = {
  className: _propTypes["default"].string,
  children: _propTypes["default"].node,
  component: _propTypes["default"].elementType,
  active: _propTypes["default"].bool,
  hasDropdown: _propTypes["default"].bool,
  hasDropup: _propTypes["default"].bool,
  hoverable: _propTypes["default"].bool,
  expanded: _propTypes["default"].bool,
  tab: _propTypes["default"].bool
};
NavbarItem.defaultProps = {
  className: undefined,
  children: null,
  component: 'div',
  active: false,
  hasDropdown: false,
  hasDropup: false,
  hoverable: false,
  expanded: false,
  tab: false
};
